packages/openid4vc-holder/src/issuance/OpenId4VciHolderService.ts

```diff
@@ -0,0 +1,703 @@
+        const { format, types } = offeredCredentialWithMetadata
+        // only requests credentials with the exact same set of types and format
+        return ctr.format === format && ctr.types.sort().join(',') === types.sort().join(',')
+      })
+
+      if (!credentialToRequest)
+        throw new AriesFrameworkError(
+          `Could not find the the requested credential with format '${ctr.format}' and types '${ctr.types}' in the offered credentials.`
+        )
+
+      return credentialToRequest
+    })
+
+    const receivedCredentials: W3cCredentialRecord[] = []
+
+    // Loop through all the credentialTypes in the credential offer
+    for (const credentialWithMetadata of credentialsToRequestWithMetadata ?? offeredCredentialsWithMetadata) {
+      // Get all options for the credential request (such as which kid to use, the signature algorithm, etc)
+      const { verificationMethod, signatureAlgorithm } = await this.getCredentialRequestOptions(agentContext, {
+        allowedCredentialFormats: supportedCredentialFormats,
+        allowedProofOfPossessionSignatureAlgorithms,
+        offeredCredentialWithMetadata: credentialWithMetadata,
+        proofOfPossessionVerificationMethodResolver,
+      })
+
+      const callbacks: ProofOfPossessionCallbacks<unknown> = {
+        signCallback: this.signCallback(agentContext, verificationMethod),
+      }
+
+      // Create the proof of possession
+      const proofInput = await ProofOfPossessionBuilder.fromAccessTokenResponse({
+        accessTokenResponse: accessToken,
+        callbacks,
+        version,
+      })
+        .withEndpointMetadata(metadata)
+        .withAlg(signatureAlgorithm)
+        .withClientId(verificationMethod.controller)
+        .withKid(verificationMethod.id)
+        .build()
+
+      this.logger.debug('Generated JWS', proofInput)
+
+      // Acquire the credential
+      const credentialRequestBuilder = new CredentialRequestClientBuilder()
+      credentialRequestBuilder
+        .withVersion(version)
+        .withCredentialEndpoint(metadata.credential_endpoint)
+        .withTokenFromResponse(accessToken)
+
+      const isInlineOffer = credentialWithMetadata.offerType === OfferedCredentialType.InlineCredentialOffer
+      const format = credentialWithMetadata.format
+      const originalFormat = getFormatForVersion(format, version)
+
+      let credentialTypes: string | string[]
+      if (version < OpenId4VCIVersion.VER_1_0_11) {
+        if (isInlineOffer) throw new AriesFrameworkError(`Inline credential offers not supported for version < 11`)
+        if (!credentialWithMetadata.credentialSupported.id) {
+          throw new AriesFrameworkError( // This should not happen
+            `No id provided for a credential supported entry in combination with the OpenId4VCI v8 draft`
+          )
+        }
+        credentialTypes = credentialWithMetadata.credentialSupported.id.split(`-${originalFormat}`)[0]
+      } else {
```

Comment:
There is a check ensuring that the version is smaller then 11 right?
If the version is 11 this should not be required as the metadata is structured differently.

Reply:
Yes there is a check for smaller than 11. But if it is larger than 11 we will enter the `else` statement. If we have a newer version, let's say `13`, we will also enter the else statement, even if that only has worked for `11`.

Reply:
thanks

Reply:
Okay, now I understand. In my opinion, this is the best behavior for now. Since in the case of v > 11, let's say 13, I want to set the behavior of v11 as the default, as this is more likely to be the case right?

Reply:
IMO it would be better to error out as a lot of undefined behavior can happen because of this. It is exactly "likely" that it is correct, but not 100%.

Reply:
I think almost all code has a fallback to v11 so it does not really matter if we change this here, you can leave it as-is.